Content.Server/Devour/DevourSystem.cs

```diff
@@ -1,49 +1,89 @@
-using Content.Server.Body.Systems;
-using Content.Shared.Chemistry.Components;
+using Content.Server.Body.Components;
+using Content.Shared.Damage.Components;
 using Content.Shared.Devour;
 using Content.Shared.Devour.Components;
 using Content.Shared.Humanoid;
+using Content.Shared.Mobs.Components;
 
 namespace Content.Server.Devour;
 
 public sealed class DevourSystem : SharedDevourSystem
 {
-    [Dependency] private readonly BloodstreamSystem _bloodstreamSystem = default!;
-
     public override void Initialize()
     {
         base.Initialize();
 
         SubscribeLocalEvent<DevourerComponent, DevourDoAfterEvent>(OnDoAfter);
+        SubscribeLocalEvent<DevourerComponent, BeingGibbedEvent>(OnBeingGibbed);
+        SubscribeLocalEvent<DevourerComponent, ComponentRemove>(OnRemoved);
     }
 
     private void OnDoAfter(EntityUid uid, DevourerComponent component, DevourDoAfterEvent args)
     {
         if (args.Handled || args.Cancelled)
             return;
 
-        var ichorInjection = new Solution(component.Chemical, component.HealRate);
+        //Store the target in the stomach if it's allowed.
+        if (component.ShouldStoreDevoured && HasComp<MobStateComponent>(args.Args.Target))
+        {
+            ContainerSystem.Insert(args.Args.Target.Value, component.Stomach);
+
+            EnsureComp<DevouredComponent>(args.Args.Target.Value);
+
+            var ev = new DevouredEvent(uid);
+            RaiseLocalEvent(args.Args.Target.Value, ref ev);
+        }
 
+        //Only alter the passive healing if the devoured entity belongs to the devourers food preference.
         if (component.FoodPreference == FoodPreference.All ||
-            (component.FoodPreference == FoodPreference.Humanoid && HasComp<HumanoidAppearanceComponent>(args.Args.Target)))
+            (component.FoodPreference == FoodPreference.Humanoid &&
+             HasComp<HumanoidAppearanceComponent>(args.Args.Target)))
         {
-            ichorInjection.ScaleSolution(0.5f);
-
-            if (component.ShouldStoreDevoured && args.Args.Target is not null)
+            if(TryComp<PassiveDamageComponent>(uid, out var passiveHealing))
             {
-                ContainerSystem.Insert(args.Args.Target.Value, component.Stomach);
+                if (component.PassiveDevourHealing != null)
+                {
+                    passiveHealing.Damage += component.PassiveDevourHealing;
```

Comment:
how many bodies do a dragon stomach fit?

Reply:
i'm like 90% sure the dragon doesn't have a coded limit but if you devour the integer limit's worth of bodies it'll probably cause a server crash

Reply:
then either there needs to be a cap in the stomach or in the healing factor